packages/terra-site/src/ThemeProvider.jsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+  /**
+   * The component(s) that will be wrapped by <ThemeProvider />
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Name of class for specified theme. e.g `terra-OSC-theme`
+   */
+  themeName: PropTypes.string.isRequired,
+};
+
+const ThemeProvider = ({ themeName, children, ...customProps }) => {
+  return <div className={themeName}>{children}</div>;
```

Comment:
Agreed with not having consumers aware of class names, but I don't know if this approach resolves that though, as themeName corresponds to a class name. I'll see if I can think of other approaches.